sonar-helm-for-iac/build.gradle.kts

```
@@ -77,6 +78,19 @@ if (isCi) {
     tasks.named("test") {
         dependsOn("testGoCode")
     }
+
+    // spotless is enabled only for CI, because spotless relies on Go installation being available on the machine and not in a container
+    spotless {
+        go {
+            val goVersion = providers.environmentVariable("GO_VERSION").getOrElse("1.21.1")
+            gofmt("go$goVersion")
+            target("**/*.go")
+            targetExclude("**/*.pb.go")
+        }
+    }
+    tasks.named("check") {
+        dependsOn("spotlessCheck")
+    }
```

I would suggest adding something like this (maybe changing the message). This way we could direct the user to a tool which he should run locally.

```suggestion
    // spotless is enabled only for CI, because spotless relies on Go installation being available on the machine and not in a container
    spotless {
        go {
            val goVersion = providers.environmentVariable("GO_VERSION").getOrElse("1.21.1")
            gofmt("go$goVersion")
            target("**/*.go")
            targetExclude("**/*.pb.go")
        }
    }
    tasks.named("check") {
        dependsOn("spotlessCheck")
    }
    tasks.named("spotlessGoCheck") {
        finalizedBy("spotlessGoCheckFailure")
    }

    tasks.register("spotlessGoCheckFailure") {
        doLast {
            println("Go code is not formatted correctly. Please format your code locally, for example with the IntelliJ Go plugin, which contains \"gofmt\".")
        }
        onlyIf {
            tasks.getByPath("spotlessGoCheck").state.failure != null
        }
    }
```

---

Or if you don't like the solution, add a bit more commentary, that gofmt is integrated into the IntelliJ Go plugin